[CI] DefaultSearchContextTests testSearchConcurrencyDoesNotCreateMoreTasksThanThreads failing

**Build Scans:**
- [elasticsearch-periodic #4665 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/cjn3f5d5zu4ag)
- [elasticsearch-periodic #4660 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/zvzmq4mfcgdne)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.DefaultSearchContextTests.testSearchConcurrencyDoesNotCreateMoreTasksThanThreads" -Dtests.seed=B9D562CBFA07B563 -Dtests.locale=rwk -Dtests.timezone=Asia/Ashgabat -Druntime.java=21
```

**Applicable branches:**
8.16

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.DefaultSearchContextTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSearchConcurrencyDoesNotCreateMoreTasksThanThreads),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a value less than <66L>
     but: <80L> was greater than <66L>
```

**Issue Reasons:**
- [8.16] 2 failures in test testSearchConcurrencyDoesNotCreateMoreTasksThanThreads (2.8% fail rate in 72 executions)
- [8.16] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/search/DefaultSearchContextTests.java
+++ b/server/src/test/java/org/elasticsearch/search/DefaultSearchContextTests.java
@@ -1056,8 +1056,9 @@ public class DefaultSearchContextTests extends MapperServiceTestCase {
         assertThat(executor.getCompletedTaskCount(), greaterThanOrEqualTo((long) numIters * executorPoolSize));
         // while we parallelize we also limit the number of tasks that each searcher submits
         assertThat(executor.getCompletedTaskCount(), lessThan((long) numIters * numSegmentTasks));
-        // *3 is just a wild guess to account for tasks that get executed while we are still submitting
-        assertThat(executor.getCompletedTaskCount(), lessThan((long) numIters * executorPoolSize * 3));
+        // *10 is just a wild guess to account for tasks that get executed while we are still submitting. The important bit is that it is a
+        // much lower factor than the number of segments
+        assertThat(executor.getCompletedTaskCount(), lessThan((long) numIters * executorPoolSize * 10));
     }
 
     private void doTestSearchConcurrency(ThreadPoolExecutor executor, int numIters, int numSegmentTasks, AtomicInteger completedTasks)

